dask_planner/src/sql/logical/table_scan.rs

```diff
@@ -16,6 +17,73 @@ pub struct PyTableScan {
     input: Arc<LogicalPlan>,
 }
 
+#[pyclass(name = "FilteredResult", module = "dask_planner", subclass)]
+#[derive(Debug, Clone)]
+pub struct PyFilteredResult {
+    // Certain Expr(s) do not have supporting logic in pyarrow for IO filtering
+    // at read time. Those Expr(s) cannot be ignored however. This field stores
+    // those Expr(s) so that they can be used on the Python side to create
+    // Dask operations that handle that filtering as an extra task in the graph.
+    #[pyo3(get)]
+    pub io_unfilterable_exprs: Vec<PyExpr>,
+    // Expr(s) that can have their filtering logic performed in the pyarrow IO logic
+    // are stored here in a DNF format that is expected by pyarrow.
+    #[pyo3(get)]
+    pub filtered_exprs: Vec<(String, String, Vec<String>)>,
+}
+
+impl PyTableScan {
+    /// Transform the singular Expr instance into its DNF form serialized in a Vec instance. Possibly recursively expanding
+    /// it as well if needed.
+    pub fn _expand_dnf_filter(
+        filter: &Expr,
+    ) -> Result<Vec<(String, String, Vec<String>)>, DaskPlannerError> {
+        let mut filter_tuple: Vec<(String, String, Vec<String>)> = Vec::new();
+
+        match filter {
+            Expr::InList {
+                expr,
+                list,
+                negated: _,
+            } => {
+                let il: Vec<String> = list.iter().map(|f| f.canonical_name()).collect();
+                filter_tuple.push((expr.canonical_name(), "in".to_string(), il));
```

Comment:
Does it make sense to check here if `expr` is a column or some scalar kind of type?
Can't think of a example but for cases where the expr instance maps to a binaryexpr or something that can't be applied directly, we'd want to use the unfiltered case.